### PR TITLE
Improvements to the alloc

### PR DESCRIPTION
**Pull request by ghost**
Improvements made:

- Consolidated the conditional expression for determining the index in the allocation/free size arrays.
- Utilized ternary operators for concise conditional assignments.
- Removed redundant comments to emphasize readability through code clarity.
- Simplified the memory peak update using a ternary expression.

### CODE REVIEW

**Comment by chrisbra** [comment]
thanks, but please don't change the function definition style. A function definition always starts in column 0, it makes it easy to grep for the definition of a function.

**Comment by k-takata** [comment]
Hmm, I don't see practical improvements in this PR.
Whether using conditional operators or if statements is mostly subjective issue. Recent compilers may generate the same code for both of them.

File: src/alloc.c [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 #if defined(MEM_PROFILE) || defined(PROTO)
 
-# define MEM_SIZES  8200
+#define MEM_SIZES 8200
```
Comment (k-takata) [inline src/alloc.c:22]: Please don't remove the indentation of `define`.

File: src/alloc.c [hunk 2]
```diff
@@ -28,106 +28,98 @@ static long_u mem_peak;
 static long_u num_alloc;
 static long_u num_freed;
 
-    static void
+static void
```
Comment (k-takata) [inline src/alloc.c:31]: Please don't remove the indentation.

File: src/alloc.c [hunk 3]
```diff
@@ -28,106 +28,98 @@ static long_u mem_peak;
...
 
-    static void
-mem_post_alloc(
-    void **pp,
-    size_t size)
+static void mem_post_alloc(void **pp, size_t size)
 {
     if (*pp == NULL)
-	return;
+        return;
```
Comment (k-takata) [inline src/alloc.c:45]: Please don't change the indentation from tab to space.

**Pull request closed without merging**